Split Importing content assembly (#4774)
theforeman/foreman-documentation: guides/common/modules/proc_extracting-gpg-keys-from-rpm-packages.adoc

:_mod-docs-content-type: PROCEDURE

> [id="Extracting_GPG_Keys_from_RPM_Packages_{context}"]

Reviewer: ```suggestion
[id="extracting-gpg-keys-from-rpm-packages"]
```

= Extracting GPG keys from RPM packages

[role="_abstract"]

> Some vendors ship their RPM repository GPG keys inside a package (for example, a `release` or `release rpm` package).

Reviewer: ```suggestion
Some vendors ship their RPM repository GPG public keys inside a package, for example, in a `release.rpm` package.
```

You can extract the key files without installing the package on your system.

.Prerequisites

> * You know the download URL for the RPM that ships the vendor GPG keys (often a release or repository configuration package).

Reviewer: ```suggestion
* You know the download URL for the RPM that ships the vendor GPG keys.
```

I would leave this out as it does not really add much value. An RPM is an RPM package; we don't know/care much what exactly is part of the RPM package or what it's called.

.Procedure
. Download the RPM to your local machine:
+
[options="nowrap" subs="+quotes"]
----
$ wget _http://www.example.com/9.5/example-9.5-2.noarch.rpm_
----
. Extract the RPM file without installing it:
+
[options="nowrap" subs="+quotes"]
----
$ rpm2cpio _example-9.5-2.noarch.rpm_ | cpio -idmv
----
. Locate the GPG public key files under the extracted `etc/pki/rpm-gpg/` directory, for example `etc/pki/rpm-gpg/RPM-GPG-KEY-_EXAMPLE-95_`.

> . Use the extracted key file when you import a {customgpg} key into {Project}.

Reviewer: OK. alternative: moved this under a "next step" heading.
